### PR TITLE
Allow using both deprecated and new setting

### PR DESCRIPTION
**Pull request by Asnelchristian**

### CODE REVIEW

**Comment by Asnelchristian** [comment]
this is will be useful to avoid problem like https://github.com/coala-analyzer/coala-bears/issues/802

File: coalib/bearlib/__init__.py [hunk 1]
```diff
@@ -38,6 +38,20 @@ def deprecate_settings(**depr_args):
...
+     >>> @deprecate_settings(new='old')
+     ... def run(new):
+     ...     print(new)
+     >>> run(old="Hello!", new='coala is always written with lowercase `c`.')
+     The setting `old` is deprecated. Please use `new` instead.
+     The value of `old` and `new` are conflicting. `new` will be used instead.
+     coala is always written with lowercase `c`.
+     >>> run(old='Hello!', new='Hello!')
+     The setting `old` is deprecated. Please use `new` instead.
+     Hello!
```
Comment (sils1297) [inline coalib/bearlib/__init__.py]: can you give old an old value and new a new value and make sure that the new one's used?

File: coalib/bearlib/__init__.py [hunk 2]
```diff
@@ -38,6 +38,20 @@ def deprecate_settings(**depr_args):
      >>> func(new='coala!')
      coala!
 
+     This example represents the case where the old and new settings are
+     provided to the function.
+
+     >>> @deprecate_settings(new='old')
+     ... def run(new):
+     ...     print(new)
+     >>> run(old="Hello!", new='coala is always written with lowercase `c`.')
```
Comment (sils1297) [inline coalib/bearlib/__init__.py]: add newline here

File: coalib/bearlib/__init__.py [hunk 3]
```diff
@@ -38,6 +38,20 @@ def deprecate_settings(**depr_args):
...
+     The setting `old` is deprecated. Please use `new` instead.
+     The value of `old` and `new` are conflicting. `new` will be used instead.
+     coala is always written with lowercase `c`.
```
Comment (sils1297) [inline coalib/bearlib/__init__.py]: what's this line do? I suggest doing at least a very short self review before submitting something, I usually try to do that.

File: coalib/bearlib/__init__.py (hunk 2 above)
Comment (sils1297) [inline coalib/bearlib/__init__.py]: on nm :D

File: coalib/bearlib/__init__.py (hunk 3 above)
Comment (sils1297) [inline coalib/bearlib/__init__.py]: nm

File: coalib/bearlib/__init__.py (hunk 1 above)
Comment (sils1297) [inline coalib/bearlib/__init__.py]: sorry :P

**Comment by sils1297** [comment]
ack https://github.com/coala-analyzer/coala/pull/2739/commits/e6d475de69087afa2dd2c1c4a59c319eb31e2d16

**Comment by sils1297** [comment]
@rultor merge

**Pull request merged**